client.go

```diff
@@ -21,12 +21,16 @@ const (
 	ClientKey  = "cert/client.key"
 )
 
+type Dailer func(target string) (*grpc.ClientConn, error)
+
 // PingClient sends echo requests to the ping server on demand.
 type PingClient struct {
-	Name     string
-	Delay    time.Duration
-	Limit    uint
-	sequence int64
+	Name       string
+	Delay      time.Duration
+	Limit      uint
+	sequence   int64
+	Connection *grpc.ClientConn
```

Comment:
Is there any reason to make the connection public?

Reply:
One has to be able to call `Close()` on that Connection. There could be a better solution by returning a `func` to defer when returning a new `Client`.